Microsoft.Toolkit.Uwp.UI/Extensions/Grid/GridLayoutDefinition.cs

```diff
@@ -0,0 +1,186 @@
+        /// of child elements of the grid, while the tuple value is the row index, row span value, column index and column span value of that
+        /// specific element.
+        /// </remarks>
+        private IDictionary<string, (int Row, int RowSpan, int Column, int ColumnSpan)> _cellProperties = new Dictionary<string, (int Row, int RowSpan, int Column, int ColumnSpan)>();
+
+        /// <summary>
+        /// Gets a list of ColumnDefinition objects defined for this layout.
+        /// </summary>
+        public List<ColumnDefinition> ColumnDefinitions { get; } = new List<ColumnDefinition>();
+
+        /// <summary>
+        /// Gets a list of RowDefinition objects defined for this layout.
+        /// </summary>
+        public List<RowDefinition> RowDefinitions { get; } = new List<RowDefinition>();
+
+        /// <summary>
+        /// Sets area definition for this layout.
+        /// </summary>
+        /// <remarks>
+        /// Area definition string should list the <see cref="FrameworkElement.Name"/> of Grid elements in row-major order.
+        /// Elements in same row should be separated with whitespaces, and rows should be separated with semicolons.
+        /// Row and column spans can be expressed by repeating element names.
+        /// </remarks>
+        /// <example>
+        /// A simple 2x3 grid layout:
+        /// .-----------.
+        /// | A | B | C |
+        /// |---|---|---|
+        /// | D | E | F |
+        /// '-----------'
+        /// A B C;
+        /// D E F;
+        ///
+        /// A 3x3 grid layout where the first and third rows span 3 columns:
+        /// .-----------------------.
+        /// |         header        |
+        /// |------.--------.-------|
+        /// | left | center | right |
+        /// |------'--------'-------|
+        /// |         footer        |
+        /// '-----------------------'
+        /// header header header;
+        /// left   center right;
+        /// footer footer footer;
+        ///
+        /// A 3x3 grid layout with row span and column span.
+        /// .----------------------.
+        /// |     |     header     |
+        /// | nav |--------.-------|
+        /// |     | center | right |
+        /// |-----'--------'-------|
+        /// |        footer        |
+        /// '----------------------'
+        /// nav    header header;
+        /// nav    center right;
+        /// footer footer footer;
+        ///
+        /// Incorrect usage:
+        /// A B C;
+        /// D A E;
+        /// will result in
+        /// .-----------.
+        /// |   : B | C |
+        /// |...A...|---|
+        /// | D :   | E |
+        /// '-----------'
+        /// Dotted lines are used to illustrate A, B and D children are overlapped.
+        /// This usage is not really invalid (it does not throw),
+        /// but this API is not expected to be used like this.
+        /// </example>
+        public string AreaDefinition
```

Comment:
Do we maybe want to rename this `AreaDefinitions` (plural) to align with the others, `RowDefinitions` and `ColumnDefinitions`?

Reply:
There is only one area definition per layout definition, whereas row definitions and column definitions are multiple.